src/platform/ESP32/PlatformManagerImpl.cpp

```diff
@@ -61,6 +62,7 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack(void)
 {
     SetConfigurationMgr(&ConfigurationManagerImpl::GetDefaultInstance());
     SetDiagnosticDataProvider(&DiagnosticDataProviderImpl::GetDefaultInstance());
+    SetDeviceInfoProvider(&DeviceInfoProviderImpl::GetDefaultInstance());
```

Comment:
So I guess an application could call `SetDeviceInfoProvider` itself with its own info?  But then `DeviceInfoProviderImpl` would be dead code being carted around for no reason.  The Linux/Mac impls can afford to do that carting around, maybe.  But I'm not sure we want that on esp32...